src/v/raft/append_entries_buffer.cc

```diff
@@ -22,28 +23,32 @@ append_entries_buffer::append_entries_buffer(
 
 ss::future<append_entries_reply>
 append_entries_buffer::enqueue(append_entries_request&& r) {
-    return ss::with_gate(_gate, [this, r = std::move(r)]() mutable {
-        // we normally do not want to wait as it would cause requests
-        // reordering. Reordering may only happend if we would wait on condition
-        // variable.
+    gate_guard guard(_gate);
 
-        return _flushed
-          .wait([this] { return _requests.size() < _max_buffered; })
-          .then([this, r = std::move(r)]() mutable {
-              ss::promise<append_entries_reply> p;
-              auto f = p.get_future();
-              _requests.push_back(std::move(r));
-              _responses.push_back(std::move(p));
-              _enqueued.signal();
-              return f;
-          });
-    });
+    // we normally do not want to wait as it would cause requests
+    // reordering. Reordering may only happend if we would wait on condition
+    // variable.
+
+    return _flushed.wait([this] { return _requests.size() < _max_buffered; })
+      .then([this, r = std::move(r), guard = std::move(guard)]() mutable {
+          ss::promise<append_entries_reply> p;
+          auto f = p.get_future();
+          _requests.push_back(std::move(r));
+          _responses.push_back(std::move(p));
+          _enqueued.signal();
+          // do not wait for the future to finish inside the gate
+          return f;
+      });
 }
```

Comment:
what does this change do? it looks like this has the exact same semantics as before, but you are now using a gate_guard but also didn't switch to coroutines.